fairscale/nn/misc/flatten_params_wrapper.py

```diff
@@ -131,22 +171,61 @@ def __getattr__(self, name: str) -> Any:
             return getattr(self.module, name)  # fallback to wrapped module
 
     def state_dict(self, *args: Any, **kwargs: Any) -> "OrderedDict[str, Tensor]":  # type: ignore
-        """Return an unflattened state_dict."""
-        with self.unflatten_params():
-            return self.module.state_dict(*args, **kwargs)
+        """Return the wrapped module's state_dict (unflattened)."""
+        if self.is_flattened and self._auto_unflatten_state_dict:
+            with self.unflatten_params(recurse=False):
+                return super().state_dict(*args, **kwargs)
+        else:
+            return super().state_dict(*args, **kwargs)
+
+    def _post_state_dict_hook(
```

Comment:
make this a static method too? In fact, why not make both hooks normal methods outside of the class? They don't need "self", right?